client/src/js/components/bhCronEmailReport/bhCronEmailReport.js

```diff
@@ -38,13 +39,19 @@ function bhCronEmailReportController(CronEmailReports, Notify, Session) {
   $ctrl.$onInit = init;
 
   function init() {
-    $ctrl.cron = {
-      report_id : $ctrl.reportId,
-      report_url : $ctrl.reportUrl,
-      has_dynamic_dates : 0,
-    };
-    $ctrl.isEmailFeatureEnabled = Session.enterprise.settings.enable_auto_email_report;
-    load($ctrl.reportId);
+    loadReportDetails($ctrl.reportKey)
+      .then(([report]) => {
+        $ctrl.cron = {
+          report_id : report.id,
+          has_dynamic_dates : 0,
+        };
+        $ctrl.isEmailFeatureEnabled = Session.enterprise.settings.enable_auto_email_report;
+        load(report.id);
+      });
+  }
+
+  function loadReportDetails(key) {
```

Comment:
Just curious - why create a separate function for this?  Why not just call `BaseReport.requestKey()` in the `init()` function?